tests/basic_correctness/test_basic_correctness.py

```diff
@@ -70,6 +73,65 @@ def test_models(
     )
 
 
+@multi_gpu_test(num_gpus=2)
```

Comment:
since this includes `fork_new_process_for_each_test`, can you check if it's forking a new process for every test?

it looks to me it is forking for the following suite of tests.

Reply:
Yes, this is intended as the original tests in `test_basic_distributed_correctness` also used `fork_new_process_for_each_test`.

Reply:
I mean the order of decorators.

```
@pytest.parametrize("a", [1, 2, 3])
@fork_new_process_for_each_test
def test_f(a):
    ...
```

we fork for every value of `a`

```
@fork_new_process_for_each_test
@pytest.parametrize("a", [1, 2, 3])
def test_f(a):
    ...
```

we only fork once for all 3 tests.

Reply:
I see what you mean. Let me check.

Reply:
As shown in this line, the original function is returned when the `pytest.mark` decorator is applied.

https://github.com/pytest-dev/pytest/blob/main/src/_pytest/mark/structures.py#L357

so the position of `fork_new_process_for_each_test` does not matter.

Reply:
that's surprising to me. can you check?

I don't have much knowledge about pytest, but it seems to me `@pytest.mark.parametrize` will repeat the several times with different parameters. and the order of decorators should matter.

Reply:
that's amazing!